src/main/java/com/mraof/minestuck/block/EnumKeyType.java

```diff
@@ -0,0 +1,36 @@
+package com.mraof.minestuck.block;
+
+import net.minecraft.util.IStringSerializable;
+
+public enum EnumKeyType implements IStringSerializable
+{
+	none,
+	tier_1_key,
+	tier_2_key;
+	
+	@Override
+	public String getName()
+	{
+		return name().toLowerCase();
+	}
+	
+	public static EnumKeyType fromString(String str) //converts string back into enum, used primarily in Dungeon Interface block/tile entity
+	{
+		for(EnumKeyType type : EnumKeyType.values())
+		{
+			if(type.toString().equals(str))
+				return type;
+		}
+		return null;
+	}
+	
+	public static EnumKeyType fromInt(int ordinal) //converts int back into enum, used primarily in KeyItem
+	{
+		for(EnumKeyType type : EnumKeyType.values())
+		{
+			if(type.ordinal() == ordinal)
+				return type;
+		}
+		return null;
+	}
```

Comment:
Would it be smarter to work with a single conversion type for storing nbt as opposed to switching between instances of strings or ints with the enum?

Reply:
Yeah, I think so